[internal/memorylimiter] bug with gopsutil v4 causes segfault on gotestsum

### Component(s)

processor/memorylimiter, service

### What happened?

**Describe the bug**
when running `make gotest` on latest branch on MacOS Arm environment (M3 Max with v15.4.1), running `go test -v ./... -race` in `processor/memorylimiter` causes segmentation fault

**Steps to reproduce**
* MacOS 15.4.1, ARM processor
* download opentelemetry-collector main branch
* run "make gotest"


**What did you expect to see?**
All tests pass

**What did you see instead?**
```
signal: segmentation fault
FAIL    go.opentelemetry.io/collector/processor/memorylimiterprocessor  0.934s
=== RUN   TestProviders
--- PASS: TestProviders (0.00s)
=== RUN   TestNewTelemetryBuilder
--- PASS: TestNewTelemetryBuilder (0.00s)
PASS
ok      go.opentelemetry.io/collector/processor/memorylimiterprocessor/internal/metadata        1.680s
=== RUN   TestSetupTelemetry
--- PASS: TestSetupTelemetry (0.00s)
PASS
ok      go.opentelemetry.io/collector/processor/memorylimiterprocessor/internal/metadatatest    1.498s
FAIL
```


### Collector version

v0.125.0

### Environment information

## Environment
OS: MacOS 15.4.1
golang 1.23.8

### Additional context

appears to be a bug in `gopsutil` v4 dependency: https://github.com/shirou/gopsutil/issues/1774.
Rolling back to v4.24.8 fixes the issue, PR to be opened shortly. Have tried all the intermediate versions.

Additionally we should consider updating CI to use MacOS 15 vs. 14, will open an issue for that as well.

Also appears to be affecting `service` package, as that is the only other one that explicitly imports `gopsutil`.

It seems like the issue doesn't affect every mac; specifically my personal macbook pro with the M1 pro doesn't have the problem. I have tried everything short of wiping my laptop, including reinstalling environments, tools, clearing env vars, etc. so I don't think it's just my setup that is causing it especially with the upstream issue with `gopsutil`.

```
john.peterson@COMP-HHVKJYJXCY opentelemetry-collector % make -C extension/memorylimiterextension test
/Users/john.peterson/otel/opentelemetry-collector/.tools/gotestsum --packages="./..." -- -race -timeout 240s
∅  internal/metadata
✖  . (313ms)

=== Failed
=== FAIL: .  (0.00s)
signal: segmentation fault
FAIL    go.opentelemetry.io/collector/extension/memorylimiterextension  0.313s

DONE 0 tests, 1 failure in 0.313s
make: *** [test] Error 1
```

```
john.peterson@COMP-HHVKJYJXCY opentelemetry-collector % make -C processor/memorylimiterprocessor test
/Users/john.peterson/otel/opentelemetry-collector/.tools/gotestsum --packages="./..." -- -race -timeout 240s
∅  internal
✖  . (553ms)
✓  internal/metadatatest (1.704s)
✓  internal/metadata (1.878s)

=== Failed
=== FAIL: .  (0.00s)
signal: segmentation fault
FAIL    go.opentelemetry.io/collector/processor/memorylimiterprocessor  0.553s

DONE 3 tests, 1 failure in 1.878s
make: *** [test] Error 1
```

```
john.peterson@COMP-HHVKJYJXCY opentelemetry-collector % make -C service test                  
/Users/john.peterson/otel/opentelemetry-collector/.tools/gotestsum --packages="./..." -- -race -timeout 240s
∅  internal/builders
∅  internal/moduleinfo
∅  internal/promtest
✖  . (821ms)
✖  internal/builders/builders_test (1.214s)
✓  extensions (1.514s)
✓  internal/attribute (1.986s)
✓  internal/capabilityconsumer (2.363s)
✖  internal/proctelemetry (2.423s)
✓  internal/graph (2.516s)
✓  internal/metadatatest (2.62s)
✓  internal/status/statustest (2.805s)
✓  internal/metadata (2.995s)
✓  internal/obsconsumer (3.189s)
✓  internal/resource (3.55s)
✓  internal/status (3.738s)
✓  internal/testcomponents (3.612s)
✓  internal/zpages (3.685s)
✓  pipelines (3.858s)
✓  telemetry (3.243s)
✓  telemetry/internal/migration (3.202s)

=== Failed
=== FAIL: .  (0.00s)
signal: segmentation fault
FAIL    go.opentelemetry.io/collector/service   0.821s

=== FAIL: internal/builders/builders_test  (0.00s)
signal: segmentation fault
FAIL    go.opentelemetry.io/collector/service/internal/builders/builders_test   1.214s

=== FAIL: internal/proctelemetry  (0.00s)
signal: segmentation fault
FAIL    go.opentelemetry.io/collector/service/internal/proctelemetry    2.423s

DONE 374 tests, 3 failures in 4.859s
make: *** [test] Error 1
```

--- a/cmd/otelcorecol/go.mod
+++ b/cmd/otelcorecol/go.mod
@@ -40,7 +40,7 @@ require (
 	github.com/cenkalti/backoff/v5 v5.0.2 // indirect
 	github.com/cespare/xxhash/v2 v2.3.0 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
-	github.com/ebitengine/purego v0.8.3 // indirect
+	github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9 // indirect
 	github.com/felixge/httpsnoop v1.0.4 // indirect
 	github.com/foxboron/go-tpm-keyfiles v0.0.0-20250323135004-b31fac66206e // indirect
 	github.com/fsnotify/fsnotify v1.9.0 // indirect

--- a/extension/memorylimiterextension/go.mod
+++ b/extension/memorylimiterextension/go.mod
@@ -59,8 +59,6 @@ require (
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
-replace github.com/ebitengine/purego => github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9
-
 replace go.opentelemetry.io/collector/component => ../../component
 
 replace go.opentelemetry.io/collector/component/componenttest => ../../component/componenttest

--- a/extension/memorylimiterextension/go.mod
+++ b/extension/memorylimiterextension/go.mod
@@ -16,7 +16,7 @@ require (
 
 require (
 	github.com/davecgh/go-spew v1.1.1 // indirect
-	github.com/ebitengine/purego v0.8.3 // indirect
+	github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9 // indirect
 	github.com/go-logr/logr v1.4.2 // indirect
 	github.com/go-logr/stdr v1.2.2 // indirect
 	github.com/go-ole/go-ole v1.2.6 // indirect
@@ -58,6 +58,8 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
+
+replace github.com/ebitengine/purego => github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9
 
 replace go.opentelemetry.io/collector/component => ../../component
 

--- a/internal/e2e/go.mod
+++ b/internal/e2e/go.mod
@@ -44,7 +44,7 @@ require (
 	github.com/cenkalti/backoff/v5 v5.0.2 // indirect
 	github.com/cespare/xxhash/v2 v2.3.0 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
-	github.com/ebitengine/purego v0.8.3 // indirect
+	github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9 // indirect
 	github.com/felixge/httpsnoop v1.0.4 // indirect
 	github.com/foxboron/go-tpm-keyfiles v0.0.0-20250323135004-b31fac66206e // indirect
 	github.com/fsnotify/fsnotify v1.9.0 // indirect
@@ -146,6 +146,8 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
+
+replace github.com/ebitengine/purego => github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9
 
 replace go.opentelemetry.io/collector => ../..
 

--- a/internal/e2e/go.mod
+++ b/internal/e2e/go.mod
@@ -147,8 +147,6 @@ require (
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
-replace github.com/ebitengine/purego => github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9
-
 replace go.opentelemetry.io/collector => ../..
 
 replace go.opentelemetry.io/collector/config/configopaque => ../../config/configopaque

--- a/internal/memorylimiter/go.mod
+++ b/internal/memorylimiter/go.mod
@@ -13,7 +13,7 @@ require (
 
 require (
 	github.com/davecgh/go-spew v1.1.1 // indirect
-	github.com/ebitengine/purego v0.8.3 // indirect
+	github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9 // indirect
 	github.com/go-logr/logr v1.4.2 // indirect
 	github.com/go-logr/stdr v1.2.2 // indirect
 	github.com/go-ole/go-ole v1.2.6 // indirect
@@ -53,6 +53,8 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
+
+replace github.com/ebitengine/purego => github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9
 
 replace go.opentelemetry.io/collector/confmap => ../../confmap
 

--- a/internal/memorylimiter/go.mod
+++ b/internal/memorylimiter/go.mod
@@ -54,8 +54,6 @@ require (
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
-replace github.com/ebitengine/purego => github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9
-
 replace go.opentelemetry.io/collector/confmap => ../../confmap
 
 replace go.opentelemetry.io/collector/component => ../../component

--- a/otelcol/go.mod
+++ b/otelcol/go.mod
@@ -40,7 +40,7 @@ require (
 	github.com/cenkalti/backoff/v4 v4.3.0 // indirect
 	github.com/cespare/xxhash/v2 v2.3.0 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
-	github.com/ebitengine/purego v0.8.3 // indirect
+	github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9 // indirect
 	github.com/go-logr/logr v1.4.2 // indirect
 	github.com/go-logr/stdr v1.2.2 // indirect
 	github.com/go-ole/go-ole v1.2.6 // indirect
@@ -120,6 +120,8 @@ require (
 	google.golang.org/protobuf v1.36.6 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+replace github.com/ebitengine/purego => github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9
 
 replace go.opentelemetry.io/collector => ../
 

--- a/otelcol/go.mod
+++ b/otelcol/go.mod
@@ -121,8 +121,6 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
-replace github.com/ebitengine/purego => github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9
-
 replace go.opentelemetry.io/collector => ../
 
 replace go.opentelemetry.io/collector/service => ../service

--- a/otelcol/otelcoltest/go.mod
+++ b/otelcol/otelcoltest/go.mod
@@ -124,8 +124,6 @@ require (
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
-replace github.com/ebitengine/purego => github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9
-
 replace go.opentelemetry.io/collector/receiver => ../../receiver
 
 replace go.opentelemetry.io/collector/consumer => ../../consumer

--- a/otelcol/otelcoltest/go.mod
+++ b/otelcol/otelcoltest/go.mod
@@ -27,7 +27,7 @@ require (
 	github.com/cenkalti/backoff/v4 v4.3.0 // indirect
 	github.com/cespare/xxhash/v2 v2.3.0 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
-	github.com/ebitengine/purego v0.8.3 // indirect
+	github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9 // indirect
 	github.com/go-logr/logr v1.4.2 // indirect
 	github.com/go-logr/stdr v1.2.2 // indirect
 	github.com/go-ole/go-ole v1.2.6 // indirect
@@ -123,6 +123,8 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
+
+replace github.com/ebitengine/purego => github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9
 
 replace go.opentelemetry.io/collector/receiver => ../../receiver
 

--- a/processor/memorylimiterprocessor/go.mod
+++ b/processor/memorylimiterprocessor/go.mod
@@ -32,7 +32,7 @@ require (
 
 require (
 	github.com/davecgh/go-spew v1.1.1 // indirect
-	github.com/ebitengine/purego v0.8.3 // indirect
+	github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9 // indirect
 	github.com/go-logr/logr v1.4.2 // indirect
 	github.com/go-logr/stdr v1.2.2 // indirect
 	github.com/go-ole/go-ole v1.2.6 // indirect
@@ -94,6 +94,8 @@ retract (
 	v0.76.0 // Depends on retracted pdata v1.0.0-rc10 module, use v0.76.1
 	v0.69.0 // Release failed, use v0.69.1
 )
+
+replace github.com/ebitengine/purego => github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9
 
 replace go.opentelemetry.io/collector/pdata/pprofile => ../../pdata/pprofile
 

--- a/processor/memorylimiterprocessor/go.mod
+++ b/processor/memorylimiterprocessor/go.mod
@@ -95,8 +95,6 @@ retract (
 	v0.69.0 // Release failed, use v0.69.1
 )
 
-replace github.com/ebitengine/purego => github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9
-
 replace go.opentelemetry.io/collector/pdata/pprofile => ../../pdata/pprofile
 
 replace go.opentelemetry.io/collector/consumer/xconsumer => ../../consumer/xconsumer

--- a/service/go.mod
+++ b/service/go.mod
@@ -140,8 +140,6 @@ require (
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
-replace github.com/ebitengine/purego => github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9
-
 replace go.opentelemetry.io/collector => ../
 
 replace go.opentelemetry.io/collector/connector => ../connector

--- a/service/go.mod
+++ b/service/go.mod
@@ -64,7 +64,7 @@ require (
 	github.com/cenkalti/backoff/v4 v4.3.0 // indirect
 	github.com/cespare/xxhash/v2 v2.3.0 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
-	github.com/ebitengine/purego v0.8.3 // indirect
+	github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9 // indirect
 	github.com/felixge/httpsnoop v1.0.4 // indirect
 	github.com/foxboron/go-tpm-keyfiles v0.0.0-20250323135004-b31fac66206e // indirect
 	github.com/fsnotify/fsnotify v1.9.0 // indirect
@@ -139,6 +139,8 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	sigs.k8s.io/yaml v1.4.0 // indirect
 )
+
+replace github.com/ebitengine/purego => github.com/ebitengine/purego v0.8.4-0.20250520125605-a52bc8357ac9
 
 replace go.opentelemetry.io/collector => ../
 

